src/metabase/models/cloud_migration.clj

```diff
@@ -33,29 +33,42 @@
 (t2/deftransforms :model/CloudMigration
   {:state mi/transform-keyword})
 
-(defsetting migration-store-url
-  (deferred-tru "Store URL for migrations. Internal test use only.")
+(defsetting store-use-staging
+  (deferred-tru "If staging store should be used instead of prod. True on dev.")
+  :type       :boolean
   :visibility :internal
-  :default    (if config/is-dev?
-               "https://store-api.staging.metabase.com/api/v2/migration"
-               "https://store-api.metabase.com/api/v2/migration")
+  :default    config/is-dev?
+  :doc        false
+  :export?    false)
+
+(defsetting store-url
+  (deferred-tru "Store URL.")
+  :visibility :admin ;; should be :internal, but FE doesn't get internal settings
+  :default    (str "https://store" (when (store-use-staging) ".staging") ".metabase.com")
+  :doc        false
+  :export?    false)
+
+(defsetting store-api-url
+  (deferred-tru "Store API URL.")
+  :visibility :internal
+  :default    (str "https://store-api" (when (store-use-staging) ".staging") ".metabase.com")
   :doc        false
   :export?    false)
 
 (defsetting migration-dump-file
-  (deferred-tru "Dump file for migrations. Internal test use only.")
+  (deferred-tru "Dump file for migrations.")
   :visibility :internal
   :default    nil
   :doc        false
   :export?    false)
 
 (defsetting migration-dump-version
-  (deferred-tru "Custom dump version for migrations. Internal test use only.")
+  (deferred-tru "Custom dump version for migrations.")
   :visibility :internal
-  ;; Use a known and already released version override when in dev.
+  ;; Use a known version on staging when there's no real version.
   ;; This will cause the restore to fail on cloud unless you also set `migration-dump-file` to
   ;; a dump from that version, but it lets you test everything else up to that point works.
-  :default    (when config/is-dev? "v0.49.7")
+  :default    (when (= (config/mb-version-info :tag) "vLOCAL_DEV") "v0.50.0-RC1")
```

Comment:
To migrate to cloud you need to use a version that cloud knows about, so that the cloud instance is started with at least that version. Cloud doesn't run 50 yet because it's not released. RC1 is there though, so we can use that.

I haven't fully worked out how this should work for the general purpose dev. I mean, it can't always work - if you have migrations on your branch and there is no release yet, using an older version on cloud will fail when restoring from the dump.

But setting a known version works up to that point. I think there's more sophisticated things that can be done here, like [auto-deploying PR builds to cloud](https://www.notion.so/metabase/How-to-create-an-internal-cloud-instance-dc27b9d8dd494fc8a3abd08b9ff1e13d?pvs=4#d9258689b74748a69a4aea3ba682ab75) and using the branch name as the version. But that's a separate piece of work.